pkg/config/config.go

```diff
@@ -305,14 +314,23 @@ func (c *Config) mergeDisabled(disabled DisabledChecks) {
 }
 
 // GetExcludesAsRegularExpression returns the excludes as a combined regular expression
-func (c Config) GetExcludesAsRegularExpression() string {
+func (c *Config) GetExcludesAsRegularExpression() string {
 	if c.IgnoreDefaults {
 		return strings.Join(c.Exclude, "|")
 	}
-
 	return strings.Join(append(c.Exclude, DefaultExcludes), "|")
 }
 
+// CachedExcludesAsRegexp returns the excludes as a compiled regular expression
+// The regexp compilation is cached
+// Note: This is not thread-safe
+func (c *Config) CachedExcludesAsRegexp() *regexp.Regexp {
+	if c.excludeRegexp == nil {
+		c.excludeRegexp = regexp.MustCompile(c.GetExcludesAsRegularExpression())
+	}
+	return c.excludeRegexp
+}
```

Comment:
regexp.MustCompile could panic

So I would do this

```suggestion
func (c *Config) CachedExcludesAsRegexp() (*regexp.Regexp, error) {
	if c.excludeRegexp != nil {
		return c.excludeRegexp, nil
	}

	var err error
	c.excludeRegexp, err = regexp.MustCompile(c.GetExcludesAsRegularExpression())
	return c.excludeRegexp, err
}
```

This comes with

https://github.com/editorconfig-checker/editorconfig-checker/pull/488#discussion_r2311912094